analyses/methylation-probe-annotations/01-create-bed-files.R

```diff
@@ -0,0 +1,148 @@
+  dplyr::arrange(seqid, start, end) |> 
+  ungroup()
+minus_three_utrs <-gencode_bed |> 
+  dplyr::filter(type == "three_prime_UTR", strand == "-") |> 
+  dplyr::group_by(seqid, type, transcript, strand) |> 
+  dplyr::summarize(start = min(start), end = min(end)) |> 
+  dplyr::arrange(seqid, start, end) |> 
+  ungroup()
+gencode_bed <- dplyr::bind_rows(exons_introns, plus_five_utrs, minus_five_utrs,
+                                plus_three_utrs, minus_three_utrs) |> 
+  dplyr::select(seqid, start, end, type, strand, transcript) |> 
+  dplyr::arrange(seqid, start, end, transcript) |> 
+  dplyr::distinct() 
+
+# extend 3'-UTR by 1kb
+gencode_bed <- gencode_bed |>
+  dplyr::mutate(start = case_when(type == "three_prime_UTR" &
+                                    strand == "-" ~ as.integer(start - 1000),
+                                  TRUE ~ as.integer(start)),
+                end = case_when(type == "three_prime_UTR" &
+                                  strand == "+" ~ as.integer(end + 1000),
+                                TRUE ~ as.integer(end)))
+
+# add promoter region - 1kb extension beyond the 5'-UTR
+promoters <- gencode_bed |>
+  dplyr::filter(type == "five_prime_UTR") |>
+  dplyr::mutate(type = "promoter",
+                start = case_when(strand == "-" ~ as.integer(end + 1), 
+                                  strand == "+" ~  as.integer(start - 1001)),
+                end = case_when(strand == "-" ~ as.integer(end + 1001),
+                                strand == "+" ~ as.integer(start + 1000)))
+
+# merge promoters to all other features
+gencode_features_bed <- gencode_bed |> 
+  dplyr::bind_rows(promoters) |> 
+  dplyr::mutate(start = case_when(start < 1 ~ as.integer(1),
+                                  TRUE ~ as.integer(start)),
+                end = case_when(end < 1 ~ as.integer(1),
+                                TRUE ~ as.integer(end))) |> 
+  dplyr::arrange(seqid, start, end, transcript) |>
+  dplyr::rename(transcript_id = transcript) |>
+  # dplyr::rename(Chromosome = seqid, Start = start, End = end, Strand = strand, 
+  #               Gene_Feature = type, transcript_id = transcript) |>
+  dplyr::left_join(gene_transcript_map, by = "transcript_id") |>
+  dplyr::distinct()
+
+# write gencode gene features bed format to file
+gencode_features_bed |> 
+  readr::write_tsv(
+    file.path(inputs_dir, 
+              "gencode.v39.primary_assembly.gene_features.annotation.bed"),
+    col_names = FALSE)
+
+# write methylation array CpG probe coordinates to bed format to file
+readr::read_csv(cpg_map, skip = 7) |> 
+  dplyr::select(CHR, MAPINFO, Name) |> 
+  dplyr::mutate(MAPINFO2 = MAPINFO) |>
+  dplyr::select(CHR, MAPINFO, MAPINFO2, Name) |>
+  tidyr::drop_na() |>
+  dplyr::mutate(CHR = paste0("chr", CHR)) |>
+  dplyr::arrange(CHR, MAPINFO) |>
+  dplyr::distinct() |>
+  readr::write_tsv(
+    file.path(inputs_dir,
+              "infinium-methylationepic-v-1-0-b5-manifest-file-grch37.bed"),
```

Comment:
this should be `"infinium-methylationepic-v-1-0-b5-manifest-file-grch38.bed"`, no?

Reply:
No. This parsing the array probe manifest, which is based on grch37. 

`infinium-methylationepic-v-1-0-b5-manifest-file-grch38.bed` was manually created from `infinium-methylationepic-v-1-0-b5-manifest-file-grch37.bed` 

This is the manual step of the module that I had mentioned to @jharenza. Initially, I hesitated to create this module because I could not find a human build assembly liftover tool that works as well as ENSEMBL Assembly Converter (only available as a web server tool) without losing any probes. 

See the note under input files in the module README

Reply:
so then does the input in line 63 of `methylation-probe-annotations/run-methyl-probe-annotation.sh` need to change to 37? It's currently *gr38.bed which is not found.

Reply:
@rjcorb , line 63 needs to remain as it is using `*gr38.bed` the liftover BED file need to intersect with gencode v39 gene features BED file that is generated by script 01. Just forgot to upload. Now uploaded to the input directory